zeroeq/http/server.cpp

```diff
@@ -416,16 +416,29 @@ class Server::Impl : public detail::Sender
         // remove leading '/'
         const auto path = message.request.path.substr(1);
 
-        if (!_methods[int(message.accessControlRequestMethod)].count(path))
+        std::string allowedMethods;
+        if (_endsWithSchema(path))
+        {
+            const auto endpoint = path.substr(0, path.find_last_of('/'));
+            if (_schemas.find(endpoint) == _schemas.end() ||
+                message.accessControlRequestMethod != Method::GET)
```

Comment:
The entire if/else if/else with two return in the middle is quite complicated... maybe extracting these some of it to a helper function could also help, something like: if (isCorsRequestForRegisteredSchema(message)) {...}